lib/view_component/compiler.rb

```diff
@@ -234,7 +234,7 @@ def compiled_template(file_path)
 
     def call_method_name(variant)
       if variant.present? && variants.include?(variant)
-        "call_#{variant}"
+        "call_#{variant.to_s.underscore}"
```

Comment:
@camertron @joelhawksley any concerns against using underscore here (I suggested it, fwiw). It lets us avoid the performance penalty of using `send`, the only downside is that it increases the chance for conflicts between variants (although I'd say it's really unlikely).

Reply:
@BlakeWilliams my gut says we should be collision-proof or detect collisions at a minimum.

Reply:
Yeah, detecting collisions feels important, but otherwise I'm ok with `underscore`ing.

Reply:
Thanks for the suggestion!

@BlakeWilliams I've updated the pull request so it detects collision. More suggestions are welcome, particularly regarding the error message.

By the way, just to confirm, I understand checking for collisions would have a smaller performance penalty than using `send` because collisions would be detected when pre-compling the templates while `send` would be executed at runtime. Is that correct?